src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java

```diff
@@ -104,16 +104,20 @@ static ResourceAccessManager lookupResourceAccessManager() throws Exception {
         List<ResourceAccessManager> managers =
                 GeoServerExtensions.extensions(ResourceAccessManager.class);
         ResourceAccessManager manager = null;
-        if (managers.size() == 1) {
+        int size = managers.size();
+        if (size == 1) {
             manager = managers.get(0);
         } else {
             for (ResourceAccessManager resourceAccessManager : managers) {
-                if (!(resourceAccessManager instanceof DefaultResourceAccessManager)) {
+                if (!DefaultResourceAccessManager.class.equals(resourceAccessManager.getClass())) {
                     manager = resourceAccessManager;
                     break;
                 }
             }
         }
+        // should never happen,just in case we have multiple singleton beans
+        // of type DefaultResourceAccessManager
+        if (manager == null) manager = managers.get(size - 1);
```

Comment:
Maybe better to use managers.get(0), if multiple managers are registered the first one should be the one with the highest priority (if the ExtensionPriority interface has been used)

Reply:
fixed